mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_key_test.go

```diff
@@ -0,0 +1,52 @@
+package resourcemanager_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccEphemeralServiceAccountKey_basic(t *testing.T) {
+	t.Parallel()
+
+	serviceAccount := envvar.GetTestServiceAccountFromEnv(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEphemeralServiceAccountKey_setup(serviceAccount),
```

Comment:
The reason for the setup step is to have a key already be created in order to test the ability to open an existing ephemeral service_account_key resource. The key must already be present in the project and specified in the `name` field of the ephemeral.

Having this step ensures that we can:

1. Have a service account key create that comes form the service account env
2. Obtain the service account key by providing the made key from the previous step into the `name` field (`google_service_account_key.key.name` is used as the input which was provisioned in the setup step)

Reply:
Makes sense!

I'd recommend having the setup step provisioning a new service account that the key is made for; I don't think we want to make extra keys for the test runner service account as it has a lot of permissions. Making keys for a service account with no permissions is better from a Security perspective, IMO

Reply:
good call, i'll replace with bootstrap instead of the test runner service account.